File: src/test/java/com/kata/board/entity/PostTest.java
package com.kata.board.entity;

import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
import static org.junit.jupiter.api.Assertions.*;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class PostTest {

    @Test
    @DisplayName("조회 수가 0보다 작은 경우 인스턴스 생성 예외")
    void Should_throwException_When_viewCountNegativeValueOrNull() {
        assertThatExceptionOfType(IllegalStateException.class)
            .isThrownBy(() -> Post.builder()
                .user(User.builder().build())
                .view(-5)
                .content("게시글 본문")
                .title("제목")
                .build())
            .withMessage("조회 수는 0보다 작을 수 없습니다.");
    }

}

File: src/test/java/com/kata/board/post/domain/PostTest.kt
package com.kata.board.post.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class PostTest {

    @Test
    @DisplayName("게시글 업데이트")
    fun `Should update post When normal Request`() {
        val post = Post("title", "content")
        post.update("제목", "내용")
        assertThat(post.title).isEqualTo("제목")
        assertThat(post.content).isEqualTo("내용")
    }
}
